lib/middleware/auth.ts

```diff
@@ -0,0 +1,35 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+// Handles client auth operations.
+// Currently we only have authorization (e.g. Identity management), associating the identities with
+// client certs.
+
+import blpapi = require('blpapi');
+import _ = require('lodash');
+import restify = require('restify');
+import Interface = require('../interface');
+import Map = require('../util/map');
+
+var sIdentityMap: Map<blpapi.Identity> = new Map<blpapi.Identity>();
+
+export function getIdentity(req: Interface.IOurRequest,
+                            res: Interface.IOurResponse,
+                            next: restify.Next): void
+{
+    if (_.has(req, 'clientCert')) {
+        var identity = sIdentityMap.get(req.clientCert.fingerprint);
+        if (identity) {
+            req.identity = identity;
```

Comment:
I think it is equivalent to say `req.identity = sIdentityMap.get(req.clientCert.fingerprint);` here. If we want to differentiate when there's no identify object, we should set a `null` here explicitly.